src/map/Map.ts

```diff
@@ -0,0 +1,123 @@
+import Phaser from "phaser";
+
+const ROOT_MAP_FOLDER = 'map';
+
+const TILE_WIDTH = 32;
+const TILE_HEIGHT = 32;
+
+const TILE_MARGIN = 0;
+const TILE_SPACING = 0;
+
+const MAP_KEY = ROOT_MAP_FOLDER;
+const TILE_SHEET_KEY = 'tileSheet';
+const TILE_SHEET_NAME = 'tiles'; // name of the tiles in the Tiled programme
+
+const MAP_DATA_FILE = 'mapData.json';
+const TILE_SHEET_FILE_PATH = 'tileset.png';
+
+const mapConfig : any = {
+  layers : {
+    'Background' : {
+      depth: 0,
+    }, 
+    'Solidground' : {
+      collisions: true,
+      depth: 10,
+      collisionsCategory: 101,
+    },
+    'Forground' : {
+      depth: 20,
+    }
+  }
+};
+
+class Map {
+
+  private scene: Phaser.Scene;
+
+  private map: Phaser.Tilemaps.Tilemap | undefined;
+
+  private tileset: Phaser.Tilemaps.Tileset | null | undefined;
+
+  public layers: any = {};
+
+  public spawners: any = {};
+
+  public height: number = 0;
+
+  public width: number = 0;
+
+  public x: number = 0;
+
+  public y: number = 0;
+  
+  constructor(scene: Phaser.Scene) {
+    this.scene = scene;
+  }
+  
+  preload(): void {
+    this.scene.load.image(TILE_SHEET_KEY, this.getFilePath(TILE_SHEET_FILE_PATH));
+    this.scene.load.tilemapTiledJSON(MAP_KEY, this.getFilePath(MAP_DATA_FILE));
+  }
```

Comment:
this is going to need to be `static`, so that it can be called without instantiating the `Map` class